run/hello-broken/Dockerfile

```diff
@@ -0,0 +1,48 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START run_broken_dockerfile]
+
+# Use the offical Golang image to create a build artifact.
+# This is based on Debian and sets the GOPATH to /go.
+# https://hub.docker.com/_/golang
+FROM golang:1.13 as builder
+
+# Create and change to the app directory.
+WORKDIR /app
+
+# Retrieve application dependencies.
+# This allows the container build to reuse cached dependencies.
+COPY go.* ./
```

Comment:
Might be more clear to be explicit with go.mod and go.sum?

Reply:
1. Ambiguous but no errors if go.mod or go.sum are missing:
```Dockerfile
COPY go.* ./
```

2. Explicit but error if go.sum is missing, such as for a simple service:
```Dockerfile
COPY go.mod go.sum ./
```

3. Partially obfuscated but no error if go.sum is missing:
```Dockerfile
COPY go.mod go.* ./
```

Currently I've implemented (1) because anyone that knows modules will probably understand. I prefer the universality of (1) and (3) as general-purpose templates and because it will simplify maintainability by not introducing the need to sometimes remove the `go.sum` reference in (2).

Do you have thoughts on the best balance for this?

Reply:
SGTM to go with option 1. Thanks for the explanation.

Reply:
Can add a comment `go.mod go.sum` to help declare intent.